src/main/scala/Core/Datapath.scala

```diff
@@ -112,16 +120,26 @@ class Datapath(val debug: Boolean = false)(implicit conf: FlexpretConfiguration)
   // higher priority than BRJMP
   if (conf.exceptions) {
```

Comment:
Below you have added an extra `when` comparing `nest_pc_sel_csr_addr`, can you explain why?

Reply:
Are you referring to the `when(io.control.next_pc_sel_csr_addr === CSRs.mepc.U)`?

This is part of implementing the `mret` instruction, that was not available before. The `when` is only triggered when an `mret` instruction is run. This sets the next PC to the value in the `mepc`, which is the machine exception PC (i.e., the PC we had before the exception occured). That is how we return back to where exceptions occured.